src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitTextFieldTests.cs

```diff
@@ -0,0 +1,267 @@
+                });
+
+            var bitTextField = isMultiLine ? component.Find(".bit-txt textarea") : component.Find(".bit-txt input");
+
+            bitTextField.Click();
+
+            Assert.AreEqual(isEnabled ? 1 : 0, component.Instance.CurrentCount);
+        }
+
+        [DataTestMethod,
+            DataRow(true, false),
+            DataRow(true, true),
+            DataRow(false, false),
+            DataRow(false, true),
+        ]
+        public void BitTextFieldMustRespondToTheFocusEvent(bool isEnabled, bool isMultiLine)
+        {
+            var component = RenderComponent<BitTextFieldTest>(
+                parameters =>
+                {
+                    parameters.Add(p => p.IsEnabled, isEnabled);
+                    parameters.Add(p => p.IsMultiLine, isMultiLine);
+                });
+
+            var bitTextField = isMultiLine ? component.Find(".bit-txt textarea") : component.Find(".bit-txt input");
+
+            bitTextField.Focus();
+            Assert.AreEqual(isEnabled ? 1 : 0, component.Instance.CurrentCount);
+
+            bitTextField.FocusIn();
+            Assert.AreEqual(isEnabled ? 2 : 0, component.Instance.CurrentCount);
+
+            bitTextField.FocusOut();
+            Assert.AreEqual(isEnabled ? 3 : 0, component.Instance.CurrentCount);
+        }
+
+        [DataTestMethod,
+            DataRow(true, false),
+            DataRow(true, true),
+            DataRow(false, false),
+            DataRow(false, true),
+        ]
+        public void BitTextFieldMustRespondToTheKeyEvent(bool isEnabled, bool isMultiLine)
+        {
+            var component = RenderComponent<BitTextFieldTest>(
+                parameters =>
+                {
+                    parameters.Add(p => p.IsEnabled, isEnabled);
+                    parameters.Add(p => p.IsMultiLine, isMultiLine);
+                });
+
+            var bitTextField = isMultiLine ? component.Find(".bit-txt textarea") : component.Find(".bit-txt input");
+
+            bitTextField.KeyDown("down");
+            Assert.AreEqual(isEnabled ? 1 : 0, component.Instance.CurrentCount);
+
+            bitTextField.KeyUp("up");
+            Assert.AreEqual(isEnabled ? 2 : 0, component.Instance.CurrentCount);
+        }
+
+        [DataTestMethod,
+           DataRow(true, false),
+           DataRow(true, true),
+           DataRow(false, false),
+           DataRow(false, true),
+       ]
+        public void BitTextFieldMustRespondToTheChangeEvent(bool isEnabled, bool isMultiLine)
+        {
+            var component = RenderComponent<BitTextFieldTest>(
+                parameters =>
+                {
+                    parameters.Add(p => p.IsEnabled, isEnabled);
+                    parameters.Add(p => p.IsMultiLine, isMultiLine);
+                });
+
+            var bitTextField = isMultiLine ? component.Find(".bit-txt textarea") : component.Find(".bit-txt input");
+
+            bitTextField.Change("change");
```

Comment:
this is not the correct way to trigger the change event.
please change the value and expect the OnChange event to be triggered.

Reply:
done